test_target_ssh_simple  started to fail with most recent git-annex

another issue which came up with recent git-annex is #6900 due to it now upgrading to v10 repo.  May be related. Started to happen recently

```
$> git grep -l FAILED.*test_target_ssh_simple | head
cron-20220726/build-ubuntu.yaml-774-841b46fb-failed/1_test-datalad (master).txt
...
```

details from `cron-20220729/build-ubuntu.yaml-777-841b46fb-failed/test-datalad (maint)/14_Run datalad tests.txt`

```
2022-07-29T04:34:13.9422697Z =================================== FAILURES ===================================
2022-07-29T04:34:13.9422988Z ____________________________ test_target_ssh_simple ____________________________
2022-07-29T04:34:13.9423159Z
2022-07-29T04:34:13.9423392Z origin = '/tmp/datalad_temp_test_target_ssh_simple9xbtj2zm'
2022-07-29T04:34:13.9423787Z src_path = '/tmp/datalad_temp_test_target_ssh_simplesgvm6860'
2022-07-29T04:34:13.9424222Z target_rootpath = '/tmp/datalad_temp_test_target_ssh_simpleez6mvmhm'
2022-07-29T04:34:13.9424426Z
2022-07-29T04:34:13.9424522Z     @slow  # 26sec on travis
2022-07-29T04:34:13.9424799Z     @skip_if_on_windows  # create_sibling incompatible with win servers
2022-07-29T04:34:13.9425065Z     @skip_ssh
2022-07-29T04:34:13.9425279Z     @with_tempfile(mkdir=True)
2022-07-29T04:34:13.9425512Z     @with_tempfile(mkdir=True)
2022-07-29T04:34:13.9425731Z     @with_tempfile(mkdir=True)
2022-07-29T04:34:13.9426039Z     def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
2022-07-29T04:34:13.9426639Z         ca = dict(result_renderer='disabled')
...
2022-07-29T04:34:13.9461335Z             # now, push should work:
2022-07-29T04:34:13.9461589Z >           push(dataset=source, to="local_target")
2022-07-29T04:34:13.9461749Z
2022-07-29T04:34:13.9462143Z /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/distribution/tests/test_create_sibling.py:297:
...
2022-07-29T04:34:13.9508876Z         if incomplete_results:
2022-07-29T04:34:13.9509142Z             raise IncompleteResultsError(
2022-07-29T04:34:13.9509403Z                 failed=incomplete_results,
2022-07-29T04:34:13.9509697Z >               msg="Command did not complete successfully")
2022-07-29T04:34:13.9510139Z E           datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 1 failed:
2022-07-29T04:34:13.9510594Z E           [{'action': 'publish',
2022-07-29T04:34:13.9510857Z E             'hints': None,
2022-07-29T04:34:13.9511294Z E             'message': 'refs/heads/dl-test-branch->local_target:refs/heads/dl-test-branch '
2022-07-29T04:34:13.9511779Z E                        '[remote rejected] (Could not update working tree to new HEAD)',
2022-07-29T04:34:13.9512167Z E             'operations': ['remote-rejected', 'error'],
2022-07-29T04:34:13.9512591Z E             'path': '/tmp/datalad_temp_test_target_ssh_simplesgvm6860',
2022-07-29T04:34:13.9513030Z E             'refds': '/tmp/datalad_temp_test_target_ssh_simplesgvm6860',
2022-07-29T04:34:13.9513473Z E             'refspec': 'refs/heads/dl-test-branch:refs/heads/dl-test-branch',
2022-07-29T04:34:13.9513820Z E             'status': 'error',
2022-07-29T04:34:13.9514110Z E             'target': 'local_target',
2022-07-29T04:34:13.9514397Z E             'type': 'dataset'}]
2022-07-29T04:34:13.9514524Z
2022-07-29T04:34:13.9514917Z /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/interface/utils.py:426: IncompleteResultsError
```

--- a/datalad/distribution/tests/test_create_sibling.py
+++ b/datalad/distribution/tests/test_create_sibling.py
@@ -194,7 +194,7 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
             ui=have_webui())
         assert_not_in('enableremote local_target failed', cml.out)
 
-    GitRepo(target_path, create=False)  # raises if not a git repo
+    target_gitrepo = GitRepo(target_path, create=False)  # raises if not a git repo
     assert_in("local_target", source.repo.get_remotes())
     # Both must be annex or git repositories
     src_is_annex = AnnexRepo.is_valid_repo(src_path)
@@ -202,6 +202,7 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
     # And target one should be known to have a known UUID within the source if annex
     if src_is_annex:
         lclcfg = AnnexRepo(src_path).config
+        target_aversion = target_gitrepo.config['annex.version']
         # basic config in place
         eq_(lclcfg.get('remote.local_target.annex-ignore'), 'false')
         ok_(lclcfg.get('remote.local_target.annex-uuid'))
@@ -217,14 +218,26 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
     ok_(str(cm.value).startswith(
         "Target path %s already exists." % target_path))
     if src_is_annex:
-        target_description = AnnexRepo(target_path, create=False).get_description()
-        assert_not_equal(target_description, None)
-        assert_not_equal(target_description, target_path)
-        # on yoh's laptop TMPDIR is under HOME, so things start to become
-        # tricky since then target_path is shortened and we would need to know
-        # remote $HOME.  To not over-complicate and still test, test only for
-        # the basename of the target_path
-        ok_endswith(target_description, basename(target_path))
+        # Before we "talk" to it with git-annex directly -- we must prevent auto-upgrades
+        # since the git-annex on "remote server" (e.g. docker container) could be outdated
+        # and not support new git-annex repo version
+        target_gitrepo.config.set('annex.autoupgraderepository', "false", scope='local')
+        try:
+            target_description = AnnexRepo(target_path, create=False).get_description()
+        except CommandError as e:
+            if 'is at unsupported version' not in str(e.stderr):
+                raise
+            # we just would skip this part of the test and avoid future similar query
+            target_description = None
+        else:
+            assert target_gitrepo.config['annex.version'] == target_aversion
+            assert_not_equal(target_description, None)
+            assert_not_equal(target_description, target_path)
+            # on yoh's laptop TMPDIR is under HOME, so things start to become
+            # tricky since then target_path is shortened and we would need to know
+            # remote $HOME.  To not over-complicate and still test, test only for
+            # the basename of the target_path
+            ok_endswith(target_description, basename(target_path))
     # now, with force and correct url, which is also used to determine
     # target_dir
     # Note: on windows absolute path is not url conform. But this way it's easy
@@ -261,13 +274,15 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
             eq_(lclcfg.get('remote.local_target.push'), DEFAULT_BRANCH)
 
         # again, by explicitly passing urls. Since we are on datalad-test, the
-        # local path should work:
+        # could use local path, but then it would not use "remote" git-annex
+        # and thus potentially lead to incongruent result. So make URLs a bit
+        # different by adding trailing /. to regular target_url
         cpkwargs = dict(
             dataset=source,
             name="local_target",
             sshurl="ssh://datalad-test",
             target_dir=target_path,
-            target_url=target_path,
+            target_url="ssh://datalad-test" + target_path + "/.",
             target_pushurl="ssh://datalad-test" + target_path,
             ui=have_webui(),
         )
@@ -277,12 +292,12 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
             assert_create_sshwebserver(existing='replace', **cpkwargs)
         interactive_assert_create_sshwebserver()
 
-        if src_is_annex:
+        if src_is_annex and target_description:
             target_description = AnnexRepo(target_path,
                                            create=False).get_description()
             eq_(target_description, target_path)
 
-        eq_(target_path,
+        eq_("ssh://datalad-test" + target_path + "/.",
             source.repo.get_remote_url("local_target"))
         eq_("ssh://datalad-test" + target_path,
             source.repo.get_remote_url("local_target", push=True))

--- a/datalad/distribution/tests/test_create_sibling.py
+++ b/datalad/distribution/tests/test_create_sibling.py
@@ -277,12 +277,13 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
         # could use local path, but then it would not use "remote" git-annex
         # and thus potentially lead to incongruent result. So make URLs a bit
         # different by adding trailing /. to regular target_url
+        target_url = "ssh://datalad-test" + target_path + "/."
         cpkwargs = dict(
             dataset=source,
             name="local_target",
             sshurl="ssh://datalad-test",
             target_dir=target_path,
-            target_url="ssh://datalad-test" + target_path + "/.",
+            target_url=target_url,
             target_pushurl="ssh://datalad-test" + target_path,
             ui=have_webui(),
         )
@@ -295,9 +296,9 @@ def test_target_ssh_simple(origin=None, src_path=None, target_rootpath=None):
         if src_is_annex and target_description:
             target_description = AnnexRepo(target_path,
                                            create=False).get_description()
-            eq_(target_description, target_path)
+            eq_(target_description, target_url)
 
-        eq_("ssh://datalad-test" + target_path + "/.",
+        eq_(target_url,
             source.repo.get_remote_url("local_target"))
         eq_("ssh://datalad-test" + target_path,
             source.repo.get_remote_url("local_target", push=True))

